shared/libraries/opcuatms/opcuatms_client/src/objects/tms_client_device_impl.cpp

```diff
@@ -450,29 +451,26 @@ void TmsClientDeviceImpl::findAndCreateCustomComponents()
 
 DictPtr<IString, IFunctionBlockType> TmsClientDeviceImpl::onGetAvailableFunctionBlockTypes()
 {
-    // todo: migrate to AvailableTypes folder node
+    auto browser = clientContext->getReferenceBrowser();
 
-    const auto fbFolderNodeId = getNodeId("FB");
-    const auto methodNodeId = clientContext->getReferenceBrowser()->getChildNodeId(fbFolderNodeId, "_TmpGetAvailableFunctionBlockTypes");
+    const auto fbFolderNodeId = browser->getChildNodeId(nodeId, "FB");
+    const auto availableTypesId = browser->getChildNodeId(fbFolderNodeId, "AvailableTypes");
```

Comment:
Note that all of these nodes are optional, as they're not in the model/only available in the SDK. As such when connecting to server without the `AvailableTypes` and `DefaultConfig` nodes, it should not crash.

In general, this PR should be backwards compatible with the old main branch. Was this tested?

Reply:
onGetAvailableFunctionBlockTypes() threw an exception before my changes (because it was not implemented). So if the AvailableTypes node is missing we can throw an exception and we are still backwards compatible. Or we can return an empty list. Which one do you want to do?

Reply:
I'd return an empty list.